src/extension.ts

```diff
@@ -9,7 +9,7 @@
 import * as vscode from 'vscode';
 import {SettingsManager, SettingsManagerImpl} from './lib/settings';
 import * as targeting from './lib/targeting'
-import {DiagnosticManager, DiagnosticManagerImpl} from './lib/diagnostics';
+import {CodeAnalyzerDiagnostic, DiagnosticManager, DiagnosticManagerImpl} from './lib/diagnostics';
```

Comment:
This is interesting to be importing something so feature-specific here - in the future it would make sense to move all of the command logic/registration out as well.